### PR TITLE
[Claims] Improve claim validation with Merkle root verification and zero relay checks

### PR DESCRIPTION
**Pull request by red-0ne**
## Summary
Improve claim validation by adding Merkle root verification and zero relay checks

### Primary Changes:
- Add comprehensive validation for `RootHash` in claim messages, enforcing proper Merkle sum tree structure
- Skip settlement for claims with zero relays or zero compute units to prevent invalid payouts  
- Update tests to cover new validation scenarios for root hashes with various edge cases

### Secondary changes:
- Move claim validation tests to `types_test` package to avoid circular dependencies
- Add test utility for creating SMST roots with specific sums and counts
- Prepare `v0.1.13` upgrade plan

## Issue

image

## Type of change

Select one or more from the following:

- [ ] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Sanity Checklist

- [x] I have updated the GitHub Issue Metadata: `assignees`, `reviewers`, `labels`, `project`, `iteration` and `milestone`
- [ ] For docs: `make docusaurus_start`
- [x] For small changes: `make go_develop_and_test` and `make test_e2e`
- [ ] For major changes: `devnet-test-e2e` label to run E2E tests in CI
- [ ] For migration changes: `make test_e2e_oneshot`
- [x] 'TODO's, configurations and other docs

### CODE REVIEW

**Review by bryanchriswhite — APPROVED** [review]

**Comment by Olshansk** [comment]
@red-0ne Nothing blocking but left some post-merge improvement commens.

File: x/tokenomics/keeper/settle_pending_claims_test.go [hunk 1]
```diff
@@ -545,6 +546,54 @@ func (s *TestSuite) TestSettlePendingClaims_Settles_WhenAProofIsNotRequired() {
 	require.Equal(t, s.claimedUpokt, *expectedEvent.GetClaimedUpokt())
 }
 
+func (s *TestSuite) TestSettlePendingClaims_ClaimIgnored_WhenHasZeroSum() {
```
[the remark this answers is not in the document]
Comment (red-0ne) [reply x/tokenomics/keeper/settle_pending_claims_test.go:549]: Created in `x/proof/keeper/msg_server_create_claim_test.go`
